CONTRIBUTING.md

```diff
@@ -283,6 +283,7 @@ that should not happen and that a dev should go look at.
 - Daemon tests (integration tests that run a node and send it commands):
   - Daemon tests should test integration, not comprehensive functionality
   - Daemon tests should validate that their responses conform to a JSON schema
+- Tests should be placed in a separate package, and follow the naming pattern `$PACKAGE_test`. For example, a test of the chain package should live in a package named `chain_test`.
```

Comment:
LG. I think it's worth adding an escape hatch, which we will occasionally need. `In limited situations, a "white box" test may be placed in the same package as the code it tests`.